src/workerd/io/worker.h

```diff
@@ -446,34 +446,7 @@ class Worker::Isolate: public kj::AtomicRefcounted {
   // enum, which unfortunately we cannot forward-declare, ugh.
   void logMessage(jsg::Lock& js, uint16_t type, kj::StringPtr description);
 
-  class SubrequestClient final: public WorkerInterface {
-  public:
-    explicit SubrequestClient(kj::Own<const Isolate> isolate,
-        kj::Own<WorkerInterface> inner, kj::HttpHeaderId contentEncodingHeaderId,
-        RequestObserver& requestMetrics)
-        : constIsolate(kj::mv(isolate)), inner(kj::mv(inner)),
-          contentEncodingHeaderId(contentEncodingHeaderId),
-          requestMetrics(kj::addRef(requestMetrics)) {}
-    KJ_DISALLOW_COPY_AND_MOVE(SubrequestClient);
-    kj::Promise<void> request(
-        kj::HttpMethod method, kj::StringPtr url, const kj::HttpHeaders& headers,
-        kj::AsyncInputStream& requestBody, kj::HttpService::Response& response) override;
-    kj::Promise<void> connect(
-        kj::StringPtr host, const kj::HttpHeaders& headers, kj::AsyncIoStream& connection,
-        kj::HttpService::ConnectResponse& tunnel,
-        kj::HttpConnectSettings settings) override;
-    void prewarm(kj::StringPtr url) override;
-    kj::Promise<ScheduledResult> runScheduled(kj::Date scheduledTime, kj::StringPtr cron) override;
-    kj::Promise<AlarmResult> runAlarm(kj::Date scheduledTime) override;
-    kj::Promise<CustomEvent::Result> customEvent(kj::Own<CustomEvent> event) override;
-
-  private:
-    kj::Own<const Isolate> constIsolate;
-    kj::Own<WorkerInterface> inner;
-    kj::HttpHeaderId contentEncodingHeaderId;
-    kj::Own<RequestObserver> requestMetrics;
-  };
-
+  class SubrequestClient;
```

Comment:
I think all three of these are not needed, the classes are only referenced in worker.c++

Reply:
Let's save this for a follow on PR. These do require access to some private other bits in the parent class so these will require a bit more work. I'd rather do that separately.